scripts/prepare_lima.py

```diff
@@ -0,0 +1,191 @@
+"""Implementation derived from https://github.com/tloen/alpaca-lora"""
+import json
+import sys
+from pathlib import Path
+from typing import Optional
+
+from datasets import load_dataset
+import torch
+from torch.utils.data import random_split
+from tqdm import tqdm
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+from lit_gpt.tokenizer import Tokenizer
+
+DATA_REPO_ID = "GAIR/lima"
+DATA_FILE_NAME = "lima_data_cleaned_archive.json"
+DESTINATION_PATH = Path("data/lima")
+CHECKPOINT_DIR = Path("checkpoints/stabilityai/stablelm-base-alpha-3b")
+TEST_SPLIT_FRACTION = 0.1
+MASK_INPUTS = False  # as in alpaca-lora
+IGNORE_INDEX = -1
+SEED = 42
+
+
+def prepare(
+    destination_path: Path = DESTINATION_PATH,
+    test_split_fraction: float = TEST_SPLIT_FRACTION,
+    checkpoint_dir: Path = CHECKPOINT_DIR,
+    mask_inputs: bool = MASK_INPUTS,
+    seed: int = SEED,
+    include_multiturn_conversations: bool = False,
+    data_repo_id: str = DATA_REPO_ID,
+    ignore_index: int = IGNORE_INDEX,
+    max_seq_length: Optional[int] = None,
```

Comment:
What's the motivation of making the `max_seq_length` configurable here? The finetuning scripts already allow overriding the `max_seq_length` used

Reply:
Yeah, that's fair. We currently also have it in Dolly. For some reason, I didn't know that we had `--max_seq_length` script. Should we remove it from Dolly too, then?

Reply:
Yes.

Note that it's not an argument, but a global variable as described in https://github.com/Lightning-AI/lit-gpt/blob/main/tutorials/oom.md#reduce-the-models-context-length

Reply:
Oh I see, yes, I know this one then. In this case, I highly recommend using a command line argument for the max sequence length so that users can copy the commands we have in the documentation. For Dolly and LIMA, that's super important, otherwise, people will think it's an issue with the model, batch size and so forth.

I think we will get a lot of extra complaints about issues if we make this not more obvious here because the context lengths in these datasets are really large.

Reply:
I'm not against adding command line flags. But there's different opinions about this: https://github.com/Lightning-AI/lit-gpt/pull/286#issuecomment-1677427362

Reply:
Ok, so the suggested alternative would be to add a `MAX_SEQ_LENGTH = None` argument at the top of the `prepare_*.py`  files then so users can customize it there? And if it's `None`, it would be using the context length of the model?

Reply:
My original point was that I don't think the dataset preparation step needs it since it can be modified in finetuning already

Reply:
I was thinking about a scenario where that's going to be a bit tedious for users but on a second thought, I think this is going to be fine. Let's remove it from the dataset prep scripts then.